src/pytest_qgis/pytest_qgis.py

```diff
@@ -84,30 +178,96 @@ def qgis_processing(qgis_app: QgsApplication) -> None:
     """
     Initializes QGIS processing framework
     """
-    python_plugins_path = os.path.join(qgis_app.pkgDataPath(), "python", "plugins")
-    if python_plugins_path not in sys.path:
-        sys.path.append(python_plugins_path)
-    from processing.core.Processing import Processing
+    _initialize_processing(qgis_app)
 
-    Processing.initialize()
 
+@pytest.fixture()
+@clean_qgis_layer
+def qgis_natural_earth_countries() -> QgsVectorLayer:
+    """
+    Natural Earth countries in 50m resolution as QgsVectorLayer
 
-def pytest_configure(config: "Config") -> None:
-    """Initializes qgis session for all tests"""
+    Made with Natural Earth. Free vector and raster map data @ naturalearthdata.com
+    """
+    path_to_data = Path(__file__).parent / "data" / "ne_50m_admin_0_countries.geojson"
+    assert path_to_data.exists(), path_to_data
+    ne_countries = QgsVectorLayer(str(path_to_data), "Natural Earth Countries", "ogr")
+    assert ne_countries.isValid()
+    return ne_countries
+
+
+@pytest.fixture(autouse=True)
+def qgis_show_map(
+    qgis_app: QgsApplication,
+    qgis_iface: QgisInterface,
+    qgis_parent: QWidget,
+    qgis_natural_earth_countries: QgsVectorLayer,
+    tmp_path: Path,
+    request: "SubRequest",
+) -> None:
+    """
+    Shows QGIS map if qgis_show_map marker is used.
+    """
+    show_map_marker = request.node.get_closest_marker(SHOW_MAP_MARKER)
+    common_settings: Settings = request.config._plugin_settings  # type: ignore
+
+    if (
```

Comment:
`if show_map_marker:` could be extracted from these three ifs so it won't be tested three times.